### PR TITLE
BUG: Include datetime microseconds in subsol calculations

### LINKED ISSUE #81: BUG: discrete output due to neglecting microseconds
**The Bug**

In `apex.helpers.subsol`, specifically [these lines](https://github.com/aburrell/apexpy/blob/develop/src/apexpy/helpers.py#L165-L177), microseconds in the input datetime objects are ignored. This leads to strange discrete behaviour in the output of functions that use `subsol`, such as `mlt2mlon`.

**Strange Discrete Behaviour**
If you increment a datetime object by less than 1 second, the output of mlt2mlon is identical:

    import apexpy
    from datetime import datetime, timedelta

    dtime = datetime(2021,11,20,12,12,12)
    apex = apexpy.Apex(dtime)
    print("Some mlon value:", apex.mlt2mlon(24,dtime))
    print("Some another mlon value, <1 second away:", apex.mlt2mlon(24,dtime+timedelta(seconds=0.5)))

gives output like this:

    Some mlon value: 243.2951431274414
    Some another mlon value, <1 second away: 243.2951431274414

**Expected behavior**
The expected output is:

    Some mlon value: 243.2951431274414
    Some another mlon value, <1 second away: 243.29313278198242

**Additional context**
I encountered this problem while using Newton's method to solve for the UT time corresponding to mlt=24. The discrete behaviour results in false convergence.

### PR DESCRIPTION
**Pull request by asreimer**
Fixes #81

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
This code can be tested with the following:

    import apexpy
    from datetime import datetime, timedelta
    
    dtime = datetime(2021,11,20,12,12,12)
    apex = apexpy.Apex(dtime)
    print("Some mlon value:", apex.mlt2mlon(24,dtime))
    print("Some another mlon value, <1 second away:", apex.mlt2mlon(24,dtime+timedelta(seconds=0.5)))

which should produce the expected output:

    Some mlon value: 243.2951431274414
    Some another mlon value, <1 second away: 243.29313278198242

### CODE REVIEW

**Comment by asreimer** [comment]
I'm not sure why 2 of 4 Travis CI tests failed. They are failing because all of the tests in `tests/test_cmd.py` have failed, which is unrelated to the changes I made...

**Comment by aburrell** [comment]
The problem looks like one of the DOI links in the AUTHORS file is behaving in a finicky manor.  I am attempting to rerun one of the jobs where it fails to see if it will succeed.

File: src/apexpy/helpers.py [hunk 1]
```diff
@@ -166,15 +166,16 @@ def subsol(datetime):
         year = np.asanyarray([datetime.year])
         doy = np.asanyarray([datetime.timetuple().tm_yday])
         ut = np.asanyarray([datetime.hour * 3600 + datetime.minute * 60
-                            + datetime.second])
+                            + datetime.second + datetime.microsecond / 1e6])
```
Comment (aburrell) [inline src/apexpy/helpers.py:169]:
```suggestion
                            + datetime.second + datetime.microsecond / 1.0e6])
```

To make clear that this should be float division and not integer division.

**Comment by aburrell** [comment]
Ok, not quite sure what to do about the "broken" link (it works on the other versions and when I try it) or the failing tests for test_cmd in 3.8 on Travis.  They are passing locally with the same version.  Will have to think about this.

**Comment by aburrell** [comment]
Ok, now just the confusing unit test failures remain.

**Comment by aburrell** [comment]
I sent a message to Travis for help.  It's only the tests that create a file that are failing, but they should work since they work in other python versions on Travis.

**Comment by aburrell** [comment]
Ok, it turns out the test problem happened because pytest stopped supporting the old directory structure.  That's been updated here along with a few other things.  So hopefully the new CI will pass now!

**Review by aburrell — APPROVED** [review]

**Pull request merged**